app/controllers/application_controller.rb

```diff
@@ -1,24 +1,34 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
-    # Prevent CSRF attacks by raising an exception.
-    # For APIs, you may want to use :null_session instead.
-    protect_from_forgery with: :exception
-    before_action :configure_permitted_parameters, if: :devise_controller?
+  include Pundit
 
-    protected
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
-    def after_sign_in_path_for(_users)
-      admin_dashboard_path
-    end
+  rescue_from Pundit::NotAuthorizedError, with: :render_unauthorized
```

Comment:
This is the other meaningful change in this file - if any of our controller actions raise a `Pundit::NotAuthorizedError`, our controller will return a 401 unauthorized status. We can change this behavior (e.g. redirect instead) but I figured this would be a good place to start.